### ISSUE TITLE
Loss Function "ipcwls" in GradientBoostingSurvivalAnalysis leads to error

### ISSUE THREAD

**Issue by Soham2112** [sebp/scikit-survival#104]
Hi

I was trying to train a time-to-failure model using machine sensor data. I chose the loss function 'ipcwls' which as per the docs weights the observations by their censoring weights. Although I'm not aware of the thoery behind it, it seemed like a reasonable choice. But, the code fails while applying the fit() function with the error message 
"i**nput contains nan infinity or a value too large for dtype float64**"

FYI, All of my X variables are scaled and they take continuous values within +-50 range. Quite a few has small values close to zero (5-6 decimal places). Is the loss function choice leading to a division by zero situation? Need some clarity on this and when this loss function should not be used.

Thanks,
Soham

**Comment by sebp** [comment]
It seems to be caused by scikit-learn's check_arrays function.

Can you please provide a full trackeback to see were exactly the error 
is occurring.

**Comment by Soham2112** [comment]
Following is the screenshot of the error message I'm getting. It's happening for both 'squared' and 'ipcwls' loss functions. I want to use any one of them as they can output time-to-event which I want rather than survival curves which I'll get using 'coxph' loss.

Any way to solve this issue?
![Screen Shot 2020-04-09 at 6 13 45 PM](https://user-images.githubusercontent.com/63403853/78896699-80055800-7a8e-11ea-8a17-98876845e95b.png)

**Comment by sebp** [comment]
The invalid value is in the observed time y, not X, check with
np.isfinite(y)

**Comment by Soham2112** [comment]
Hi

np.isfinite(y) is showing incompatible. I am attaching a screenshot of the error. I have checked the dataframe of Y (n_samples,2) which has no Null/NaN values.

To. be clear, I am creating 'y' by passing the dataframe with 2 columns through the helper function **from_dataframe** which converts it into a structured array. The event indicator is binary 0/1 with **int64** dtype while the time to failure are in days but in **float64**. Are the data types the cause of the problem since there are no missing values anywhere?

Thanks
Soham
![Screen Shot 2020-04-09 at 8 23 40 PM](https://user-images.githubusercontent.com/63403853/78909066-94eae700-7aa0-11ea-87c6-06bd9d6e6977.png)

**Comment by Soham2112** [comment]
Hi @sebp , please let me know whether my approach is faulty or this is a package related issue?
I have re-checked my dataset again. There is no missing value or NaN present either in the X or y. 

Thanks again
Soham

**Comment by sebp** [comment]
Could it be that some survival time is zero or negative? This would explain NaN when taking the log.

**Comment by Soham2112** [comment]
Hi @sebp 

You were right. The problem was there were few zeros in the time-to-failure time. Now it's solved. Thanks a lot for your help btw!

**Comment by jtlz2** [comment]
@sebp Thank you - I had been getting an all-NaN slice error - and this fixed it. Thanks also for such a great package!
